mubench.pipeline/tests/tasks/implementations/test_compile.py

```diff
@@ -294,7 +294,41 @@ def test_compile_with_gradle(self, shell_mock, copy_mock):
         shell_mock.return_value = ":printClasspath\n/path/dependency1.jar\n/path/dependency2.jar\n\nBUILD SUCCESSFUL"
 
         self.uut.process_project_version(self.project, self.version)
-        assert_equals(shell_mock.mock_calls[1][1], ("gradle :printClasspath -b classpath.gradle",))
+        assert_equals(shell_mock.mock_calls[1][1], ("gradle :printClasspath -b 'classpath.gradle'",))
+        assert_in(call("/path/dependency1.jar", self.dep_path), copy_mock.mock_calls)
+        assert_in(call("/path/dependency2.jar", self.dep_path), copy_mock.mock_calls)
+
+    @patch("tasks.implementations.compile.shutil.copy")
+    @patch("tasks.implementations.compile.Shell.exec")
+    def test_compile_with_gradle_project_dir_parameter(self, shell_mock, copy_mock):
+        self.version._YAML["build"]["commands"] = ["gradle build -p /testdir/"]
+        shell_mock.return_value = ":printClasspath\n/path/dependency1.jar\n/path/dependency2.jar\n\nBUILD SUCCESSFUL"
+
+        self.uut.process_project_version(self.project, self.version)
+        assert_equals(shell_mock.mock_calls[1][1], ("gradle :printClasspath -b '/testdir/classpath.gradle'",))
+        assert_in(call("/path/dependency1.jar", self.dep_path), copy_mock.mock_calls)
```

Comment:
test_compile_with_gradle already checks that parsing the classpath from the output works. No need to repeat this in every subsequent test. Please remove the respective asserts (and the mock setup, if possible).